src/services/deployment-service/deployment-service.ts

```diff
@@ -0,0 +1,87 @@
+import Logger from "bunyan";
+import { envVars } from "config/env";
+import { getLogger } from "config/logger";
+import { dynamodb as ddb } from "config/dynamodb";
+import { hash } from "utils/hash-utils";
+
+const defaultLogger = getLogger("DeploymentDynamoLogger");
+
+const ONE_YEAR_IN_MILLISECONDS = 365 * 24 * 60 * 60 * 1000;
+
+export const saveDeploymentInfo = async (deploymentInfo : {
+	gitHubBaseUrl: string
+	repositoryId: number;
+	commitSha: string;
+	env: string;
+	createdAt: Date;
+}, logger: Logger) => {
+	logger.debug("Saving deploymentInfo to db");
+	const result = await ddb.putItem({
+		TableName: envVars.DYNAMO_DEPLOYMENT_HISTORY_CACHE_TABLE_NAME,
+		Item: {
+			Id: { "S": getKey(deploymentInfo) }, //partition key
+			CreatedAt: { "N": String(deploymentInfo.createdAt.getTime()) }, //sort key
+			CommitSha: { "S": deploymentInfo.commitSha }, //real data we need
+			ExpiredAfter: { "N": String(Math.floor((deploymentInfo.createdAt.getTime() + ONE_YEAR_IN_MILLISECONDS) / 1000)) } //ttl
+		}
+	}).promise();
+	if (result.$response.error) {
+		throw result.$response.error;
+	}
+};
+
+export type LastSuccessfulDeployment = {
+	commitSha: string;
+	createdAt: Date;
+}
+
+export const findLastSuccessDeployment = async(
```

Comment:
how aobut `findCachedLastSuccessfulDeploymentInfo`?

Reply:
Updated to `findLastSuccessDeploymentFromCache`